<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>Dialog</class>
 <widget class="QDialog" name="Dialog">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>850</width>
    <height>668</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Deform</string>
  </property>
  <layout class="QVBoxLayout">
   <property name="leftMargin">
    <number>8</number>
   </property>
   <property name="topMargin">
    <number>8</number>
   </property>
   <property name="rightMargin">
    <number>8</number>
   </property>
   <property name="bottomMargin">
    <number>8</number>
   </property>
   <item>
    <widget class="Viewer" name="viewer" native="true">
     <property name="sizePolicy">
      <sizepolicy hsizetype="Preferred" vsizetype="Preferred">
       <horstretch>0</horstretch>
       <verstretch>1</verstretch>
      </sizepolicy>
     </property>
    </widget>
   </item>
   <item>
    <layout class="QHBoxLayout">
     <item>
      <widget class="QCheckBox" name="FPSCheckBox">
       <property name="text">
        <string>FPS</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QCheckBox" name="GridCheckBox">
       <property name="text">
        <string>Grid</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QCheckBox" name="AxisCheckBox">
       <property name="text">
        <string>Axis</string>
       </property>
      </widget>
     </item>
     <item>
      <spacer>
       <property name="orientation">
        <enum>Qt::Horizontal</enum>
       </property>
       <property name="sizeHint" stdset="0">
        <size>
         <width>141</width>
         <height>31</height>
        </size>
       </property>
      </spacer>
     </item>
    </layout>
   </item>
   <item>
    <widget class="QGroupBox" name="groupBox">
     <property name="title">
      <string>Mesh</string>
     </property>
     <layout class="QGridLayout" name="gridLayout">
      <item row="1" column="0">
       <widget class="QSpinBox" name="meshRows">
        <property name="minimum">
         <number>1</number>
        </property>
        <property name="value">
         <number>12</number>
        </property>
       </widget>
      </item>
      <item row="0" column="0">
       <widget class="QLabel" name="label_2">
        <property name="text">
         <string>Rows</string>
        </property>
       </widget>
      </item>
      <item row="1" column="2">
       <widget class="QDoubleSpinBox" name="meshCellSize">
        <property name="minimum">
         <double>0.000000000000000</double>
        </property>
        <property name="singleStep">
         <double>0.100000000000000</double>
        </property>
        <property name="value">
         <double>0.100000000000000</double>
        </property>
       </widget>
      </item>
      <item row="0" column="2">
       <widget class="QLabel" name="label_3">
        <property name="text">
         <string>Cell Size</string>
        </property>
       </widget>
      </item>
      <item row="1" column="1">
       <widget class="QSpinBox" name="meshColumns">
        <property name="minimum">
         <number>1</number>
        </property>
        <property name="value">
         <number>12</number>
        </property>
       </widget>
      </item>
      <item row="0" column="1">
       <widget class="QLabel" name="label">
        <property name="text">
         <string>Columns</string>
        </property>
       </widget>
      </item>
      <item row="0" column="3">
       <widget class="QLabel" name="label_4">
        <property name="text">
         <string>Mass</string>
        </property>
       </widget>
      </item>
      <item row="1" column="3">
       <widget class="QDoubleSpinBox" name="particleMass">
        <property name="decimals">
         <number>5</number>
        </property>
        <property name="minimum">
         <double>0.000010000000000</double>
        </property>
        <property name="singleStep">
         <double>0.002500000000000</double>
        </property>
        <property name="value">
         <double>0.002500000000000</double>
        </property>
       </widget>
      </item>
     </layout>
    </widget>
   </item>
   <item>
    <widget class="QGroupBox" name="groupBox_2">
     <property name="title">
      <string>Simulator</string>
     </property>
     <layout class="QGridLayout" name="gridLayout_2">
      <item row="0" column="0">
       <widget class="QLabel" name="label_5">
        <property name="text">
         <string>Timestep</string>
        </property>
       </widget>
      </item>
      <item row="0" column="1">
       <widget class="QDoubleSpinBox" name="timestep">
        <property name="decimals">
         <number>5</number>
        </property>
        <property name="minimum">
         <double>0.000100000000000</double>
        </property>
        <property name="singleStep">
         <double>0.001000000000000</double>
        </property>
        <property name="value">
         <double>0.001000000000000</double>
        </property>
       </widget>
      </item>
      <item row="1" column="1">
       <widget class="QSpinBox" name="iterations">
        <property name="minimum">
         <number>1</number>
        </property>
        <property name="value">
         <number>10</number>
        </property>
       </widget>
      </item>
      <item row="1" column="0">
       <widget class="QLabel" name="label_6">
        <property name="text">
         <string>Iterations</string>
        </property>
       </widget>
      </item>
      <item row="2" column="0">
       <widget class="QLabel" name="label_7">
        <property name="text">
         <string>Collision Stiffness</string>
        </property>
       </widget>
      </item>
      <item row="2" column="1">
       <widget class="QDoubleSpinBox" name="collisionStiffness">
        <property name="decimals">
         <number>0</number>
        </property>
        <property name="maximum">
         <double>99999999.000000000000000</double>
        </property>
        <property name="value">
         <double>10000.000000000000000</double>
        </property>
       </widget>
      </item>
      <item row="3" column="0">
       <widget class="QLabel" name="label_8">
        <property name="text">
         <string>Spring Stiffness</string>
        </property>
       </widget>
      </item>
      <item row="3" column="1">
       <widget class="QDoubleSpinBox" name="springStiffness">
        <property name="decimals">
         <number>0</number>
        </property>
        <property name="maximum">
         <double>9999999999.000000000000000</double>
        </property>
        <property name="value">
         <double>100.000000000000000</double>
        </property>
       </widget>
      </item>
     </layout>
    </widget>
   </item>
   <item>
    <widget class="QPushButton" name="reset">
     <property name="text">
      <string>Reset</string>
     </property>
    </widget>
   </item>
  </layout>
 </widget>
 <customwidgets>
  <customwidget>
   <class>Viewer</class>
   <extends>QWidget</extends>
   <header location="global">viewer.h</header>
   <container>1</container>
   <slots>
    <slot>changeCellSize(double)</slot>
    <slot>changeMeshRows(int)</slot>
    <slot>changeMeshCols(int)</slot>
    <slot>changeParticleMass(double)</slot>
    <slot>changeTimestep(double)</slot>
    <slot>changeIterations(int)</slot>
    <slot>changeCollisionStiffness(double)</slot>
    <slot>changeSpringStiffness(double)</slot>
    <slot>reset()</slot>
   </slots>
  </customwidget>
 </customwidgets>
 <resources/>
 <connections>
  <connection>
   <sender>AxisCheckBox</sender>
   <signal>clicked(bool)</signal>
   <receiver>viewer</receiver>
   <slot>setAxisIsDrawn(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>150</x>
     <y>441</y>
    </hint>
    <hint type="destinationlabel">
     <x>150</x>
     <y>239</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>FPSCheckBox</sender>
   <signal>clicked(bool)</signal>
   <receiver>viewer</receiver>
   <slot>setFPSIsDisplayed(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>49</x>
     <y>446</y>
    </hint>
    <hint type="destinationlabel">
     <x>41</x>
     <y>263</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>viewer</sender>
   <signal>axisIsDrawnChanged(bool)</signal>
   <receiver>AxisCheckBox</receiver>
   <slot>setChecked(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>126</x>
     <y>346</y>
    </hint>
    <hint type="destinationlabel">
     <x>160</x>
     <y>446</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>viewer</sender>
   <signal>gridIsDrawnChanged(bool)</signal>
   <receiver>GridCheckBox</receiver>
   <slot>setChecked(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>80</x>
     <y>406</y>
    </hint>
    <hint type="destinationlabel">
     <x>73</x>
     <y>444</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>GridCheckBox</sender>
   <signal>clicked(bool)</signal>
   <receiver>viewer</receiver>
   <slot>setGridIsDrawn(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>108</x>
     <y>446</y>
    </hint>
    <hint type="destinationlabel">
     <x>94</x>
     <y>234</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>viewer</sender>
   <signal>FPSIsDisplayedChanged(bool)</signal>
   <receiver>FPSCheckBox</receiver>
   <slot>setChecked(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>26</x>
     <y>406</y>
    </hint>
    <hint type="destinationlabel">
     <x>29</x>
     <y>446</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>meshCellSize</sender>
   <signal>valueChanged(double)</signal>
   <receiver>viewer</receiver>
   <slot>changeCellSize(double)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>529</x>
     <y>432</y>
    </hint>
    <hint type="destinationlabel">
     <x>324</x>
     <y>173</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>meshColumns</sender>
   <signal>valueChanged(int)</signal>
   <receiver>viewer</receiver>
   <slot>changeMeshCols(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>324</x>
     <y>432</y>
    </hint>
    <hint type="destinationlabel">
     <x>324</x>
     <y>173</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>meshRows</sender>
   <signal>valueChanged(int)</signal>
   <receiver>viewer</receiver>
   <slot>changeMeshRows(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>120</x>
     <y>432</y>
    </hint>
    <hint type="destinationlabel">
     <x>324</x>
     <y>173</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>particleMass</sender>
   <signal>valueChanged(double)</signal>
   <receiver>viewer</receiver>
   <slot>changeParticleMass(double)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>555</x>
     <y>432</y>
    </hint>
    <hint type="destinationlabel">
     <x>324</x>
     <y>161</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>timestep</sender>
   <signal>valueChanged(double)</signal>
   <receiver>viewer</receiver>
   <slot>changeTimestep(double)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>477</x>
     <y>432</y>
    </hint>
    <hint type="destinationlabel">
     <x>324</x>
     <y>124</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>iterations</sender>
   <signal>valueChanged(int)</signal>
   <receiver>viewer</receiver>
   <slot>changeIterations(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>627</x>
     <y>632</y>
    </hint>
    <hint type="destinationlabel">
     <x>424</x>
     <y>207</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>springStiffness</sender>
   <signal>valueChanged(double)</signal>
   <receiver>viewer</receiver>
   <slot>changeSpringStiffness(double)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>627</x>
     <y>632</y>
    </hint>
    <hint type="destinationlabel">
     <x>424</x>
     <y>173</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>collisionStiffness</sender>
   <signal>valueChanged(double)</signal>
   <receiver>viewer</receiver>
   <slot>changeCollisionStiffness(double)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>627</x>
     <y>598</y>
    </hint>
    <hint type="destinationlabel">
     <x>424</x>
     <y>173</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>reset</sender>
   <signal>released()</signal>
   <receiver>viewer</receiver>
   <slot>reset()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>424</x>
     <y>651</y>
    </hint>
    <hint type="destinationlabel">
     <x>424</x>
     <y>156</y>
    </hint>
   </hints>
  </connection>
 </connections>
</ui>
